tools/github_update.py

```diff
@@ -172,20 +172,49 @@ def set_commit_status(self, context_label, state, message, details_url):
 class GithubStatusUpdater(object):
 
     def __init__(self, default_context_label=''):
-        info = RepoInfo()
-        self._api = GithubAPI(info.repo_orgname(), info.commit_sha(), info.github_auth_token())
+        if self._should_access_github():
+            info = RepoInfo()
+            self._api = GithubAPI(info.repo_orgname(), info.commit_sha(), info.github_auth_token())
+        else:
+            self._api = None
         self._default_context_label = default_context_label
 
 
+    def _should_access_github(self):
+        '''returns whether the local environment appears to be running in CI'''
+        if 'WORKSPACE' not in os.environ:
+            # not running in CI. skip actually sending anything to GitHub
+            return False
+        if os.environ.get('GITHUB_DISABLE', ''):
```

Comment:
Random nitpick, this `get('key', '')` pattern is pretty weird.  You set the default to the empty string a lot but never use it.
get('key') returns None if it isn't there, so this looks like you explicitly needed an empty string but don't.  

I prefer the `'WORKSPACE' in os.environ` pattern above for testing if present.

Just food for the future.